### DISCUSSION TITLE
Question about ignoring duplicate digest auth header

### DISCUSSION THREAD

**Discussion by TorkyTitor** [curl/curl#11315] [Q&A]
Hello, I am studying the part of libcurl about digest, and I saw line 1070 of curl/lib/http.c indicating that when there are duplicate digest headers, curl will ignore it. I have some confusion about this handling method.

According to rfc7616 3.9.1.  Example with SHA-256 and MD5, 2 digest header with different algorithm is in accordance with the protocol.
When encountering this situation, libcurl will process the first digest header and ignore the second digest header.

Let's assume a situation that the first digest header carries an algorithm that is not supported by libcurl and the second digest header carries an algorithm that is supported by libcurl.

**I believe the first digest header will be ignored with log "Authentication problem. Ignoring this.", but will the second digest header also be ignored with log "Ignoring duplicate digest auth header."?**

Thank you.

**Comment by bagder** [comment]
If so, that sounds like a bug.
